### PR TITLE
PAYARA-2270 Including Servlet 4 Push Code In Payara 5

### PR DESCRIPTION
**Pull request by MattGill98**
The commits show the merged changes, but in summary:
- Spdy dom element and module removed.
- HTTP2 dom element removed and functionality merged into HTTP dom element.
- Changes from domain1 copied across to payaradomain.

### CODE REVIEW

**Comment by MattGill98** [comment]
jenkins quick test payara 5 please

File: nucleus/grizzly/config/src/main/java/org/glassfish/grizzly/config/GenericGrizzlyListener.java [hunk 1]
```diff
@@ -739,95 +738,36 @@ protected void configureHttpProtocol(final ServiceLocator habitat,
...
-                // only provide the information necessary for the addon to operate.
-                // It will be important to keep this mock in sync with the details the
-                // addon requires.
-                http2Addon.setup(createMockListener(secure), builder);
-                isHttp2Enabled = true;
-            }
+        if (!skipHttp2 && httpElement != null && httpElement.isHttp2Enabled()) {
+
+            Http2AddOn http2Addon = new Http2AddOn(Http2Configuration.builder()
+                    .disableCipherCheck(httpElement.isHttp2DisableCipherCheck())
```
Comment (smillidge) [inline nucleus/grizzly/config/src/main/java/org/glassfish/grizzly/config/GenericGrizzlyListener.java]: Can these configuration properties be added to the administration console and be made configurable via an asadmin command.

**Comment by arjantijms** [comment]
jenkins quick test payara 5 please

**Comment by MattGill98** [comment]
jenkins quick test payara 5 please

**Comment by MattGill98** [comment]
Added new HTTP/2 options to admin console, create-http command as well as create-http-listener command.

**Comment by MattGill98** [comment]
Jenkins quick test payara 5 please

**Comment by MattGill98** [comment]
jenkins quick test payara 5 please

**Pull request merged**